src/e2e-test/java/io/cdap/plugin/odp/stepsdesign/RuntimeODP.java

```diff
@@ -0,0 +1,396 @@
+    }
+  }
+
+  @Then("Enter the BigQuery Properties for ODP datasource {string}")
+  public void enterTheBigQueryPropertiesForODPDatasource(String tableName) throws IOException, InterruptedException {
+    CdfStudioLocators.bigQueryProperties.click();
+    CdfBigQueryPropertiesActions cdfBigQueryPropertiesActions = new CdfBigQueryPropertiesActions();
+    SeleniumHelper.replaceElementValue(CdfBigQueryPropertiesLocators.projectID,
+                                       CDAPUtils.getPluginProp("odpProjectId"));
+    CdfBigQueryPropertiesLocators.bigQueryReferenceName.sendKeys("automation_test");
+    CdfBigQueryPropertiesLocators.datasetProjectID.sendKeys(CDAPUtils.getPluginProp("odpProjectId"));
+    CdfBigQueryPropertiesLocators.dataset.sendKeys(CDAPUtils.getPluginProp("dataSetOdp"));
+    CdfBigQueryPropertiesLocators.bigQueryTable.sendKeys(CDAPUtils.getPluginProp(tableName));
+    CdfBigQueryPropertiesLocators.truncatableSwitch.click();
+    CdfBigQueryPropertiesLocators.updateTable.click();
+    CdfBigQueryPropertiesLocators.validateBttn.click();
+    SeleniumHelper.waitElementIsVisible(CdfBigQueryPropertiesLocators.textSuccess, 1L);
+  }
+
+  @Then("Close the BQ Properties")
+  public void closeTheBQProperties() {
+    CdfGcsActions.closeButton();
+  }
+
+  @Then("Run the ODP Pipeline in Runtime")
+  public void runTheODPPipelineInRuntime() throws InterruptedException {
+    CdfPipelineRunAction.runClick();
+  }
+
+  @Then("Wait till ODP pipeline is in running state")
+  public void waitTillODPPipelineIsInRunningState() {
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 1000000);
+    wait.until(ExpectedConditions.or(
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Succeeded']")),
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Failed']"))));
+  }
+
+  @Then("Open Logs of ODP Pipeline")
+  public void openLogsOfODPPipeline() throws InterruptedException, FileNotFoundException {
+    CdfPipelineRunAction.logsClick();
+    Thread.sleep(5000); //TODO
+    rawLog = CdfPipelineRunAction.captureRawLogs();
+    BeforeActions.scenario.write(rawLog);
+    out.println(rawLog);
+    out.close();
+  }
+
+  @Then("Verify the ODP pipeline status is {string}")
+  public void verifyTheODPPipelineStatusIs(String status) {
+    boolean webelement = SeleniumHelper.verifyElementPresent("//*[@data-cy='" + status + "']");
+    Assert.assertTrue(webelement);
+  }
+
+  @Then("validate successMessage is displayed for the ODP pipeline")
+  public void validateSuccessMessageIsDisplayedForTheODPPipeline() {
+    CdfLogActions.validateSucceeded();
+  }
+
+  @Then("Get Count of no of records transferred from ODP to BigQuery in {string}")
+  public void getCountOfNoOfRecordsTransferredFromODPToBigQueryIn(String table) throws
+    IOException, InterruptedException {
+    countRecords = gcpClient.countBqQuery(CDAPUtils.getPluginProp(table));
```

Comment:
Here, we should also verify the metric in the UI, no of records out from the source to the no of records in sink.

Reply:
will fix this in CDAP-18634